### DISCUSSION TITLE
Swerve Drive

### DISCUSSION THREAD

**Discussion by adisimaimulte1** [acmerobotics/road-runner-quickstart#216] [Q&A]
I've seen that there are some classes refering to the infamous swerve, like SwerveDrive and SwerveKinematics. I'm a bit confused to how are they intended to be implemented, because I haven't seen passing in the 8 motors/servos anywhere in the source code. Can someone please explain how to use them?

**Comment by a-gt** [comment]
All those classes provide are the kinematics and stuff. It essentially just gives you the powers and angles. So you need to extend the sample drive class and implement your own classes for rotating the actual modules and such.

**Reply by PurplePizza123** [reply]
My team, Devildogs11206 https://github.com/Devildogs11206/CenterStage, used an omni drive chassis this season. Here is some basic omni code. It is not a swerve drive though.
